apps/meteor/server/services/messages/service.ts

```diff
@@ -209,4 +222,16 @@ export class MessageService extends ServiceClassInternal implements IMessageServ
 
 	// 	await Room.join({ room, user });
 	// }
+
+	async beforeReacted(message: IMessage, room: IRoom) {
+		if ((isMessageFromMatrixFederation(message) || isRoomFederated(room)) && isFederationEnabled() && !isFederationReady()) {
+			throw new FederationMatrixInvalidConfigurationError('Unable to react to message');
+		}
+	}
+
+	async beforeDelete(message: IMessage, room: IRoom) {
+		if ((isMessageFromMatrixFederation(message) || isRoomFederated(room)) && isFederationEnabled() && !isFederationReady()) {
+			throw new FederationMatrixInvalidConfigurationError('Unable to delete message');
+		}
+	}
```

Comment:
It would be good to unit-test these.

Reply:
Let me centralize these checks, then we can add tests once for that module